### PR TITLE
Fix `skaffold debug` panic with nodejs

### LINKED ISSUE #3821: skaffold debug panics on nodejs example
On examples/nodejs `skaffold debug` panics: 

<details>

```
DEBU[0005] no debug transformation for: service/node    
DEBU[0005] Found artifact for image "gcr.io/balintp-gcp-lab/node-example:076416f9c6eb2272f4716c9cc2acaf5911c4f38753120fc0d1bd34552fcb8b2b" 
DEBU[0005] Retrieved local image configuration for gcr.io/balintp-gcp-lab/node-example:076416f9c6eb2272f4716c9cc2acaf5911c4f38753120fc0d1bd34552fcb8b2b: {false false false [/bin/sh -c npm run $NODE_ENV] <nil>  [docker-entrypoint.sh] [PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin NODE_VERSION=12.16.0 YARN_VERSION=1.22.0 NODE_ENV=production]  sha256:22a37dd5d95398474e97b8dd2d497c58aafd3459a845dd68a5f6a851e76497c9 map[] [] false false false node map[] /home/node/app map[3000/tcp:{}] false false   []} 
INFO[0005] Configuring "node" for node.js debugging     
WARN[0005] Skipping "node" as does not appear to invoke node 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x4f8a068]

goroutine 1 [running]:
github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug.transformPodSpec(0xc000708a30, 0xc000708b30, 0xc000c551b8, 0x0)
        /Users/balintp/skaffold/pkg/skaffold/debug/transform.go:200 +0x288
github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug.transformManifest(0x5891260, 0xc000708900, 0xc000c551b8, 0x0)
        /Users/balintp/skaffold/pkg/skaffold/debug/transform.go:147 +0x4f2
github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug.applyDebuggingTransforms(0xc000646d50, 0x2, 0x2, 0xc000c551b8, 0xc000876200, 0x2, 0xc000438070, 0xc000c551f0, 0x4f855f3)
        /Users/balintp/skaffold/pkg/skaffold/debug/debug.go:71 +0x2b2
github.com/GoogleContainerTools/skaffold/pkg/skaffold/debug.ApplyDebuggingTransforms(0xc000646d50, 0x2, 0x2, 0xc00093b080, 0x1, 0x1, 0xc0006b0ff0, 0x0, 0x0, 0x0, ...)
        /Users/balintp/skaffold/pkg/skaffold/debug/debug.go:62 +0x174
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy.(*KubectlDeployer).renderManifests(0xc0007b7760, 0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0xc00093b080, 0x1, 0x1, 0xc000794e40, 0x4, ...)
        /Users/balintp/skaffold/pkg/skaffold/deploy/kubectl.go:260 +0x336
github.com/GoogleContainerTools/skaffold/pkg/skaffold/deploy.(*KubectlDeployer).Deploy(0xc0007b7760, 0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0xc00093b080, 0x1, 0x1, 0xc000794e40, 0x4, ...)
        /Users/balintp/skaffold/pkg/skaffold/deploy/kubectl.go:76 +0x10b
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.withTimings.Deploy(0x58ad8a0, 0xc000342420, 0x58966e0, 0xc000794e00, 0x58c9860, 0xc0007b7760, 0x1, 0x58c24a0, 0xc0000a7680, 0x5880bc0, ...)
        /Users/balintp/skaffold/pkg/skaffold/runner/timings.go:89 +0x15f
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*SkaffoldRunner).Deploy(0xc0003657a0, 0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0xc00093b080, 0x1, 0x1, 0xc00093b060, 0x1)
        /Users/balintp/skaffold/pkg/skaffold/runner/deploy.go:55 +0x32e
github.com/GoogleContainerTools/skaffold/pkg/skaffold/runner.(*SkaffoldRunner).Dev(0xc0003657a0, 0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0xc000718f80, 0x1, 0x4, 0x0, 0x0)
        /Users/balintp/skaffold/pkg/skaffold/runner/dev.go:202 +0xa4a
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doDev.func5(0x58e48c0, 0xc0003657a0, 0xc000670360, 0x0, 0x1010001000001)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/dev.go:72 +0xad
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.withRunner(0x58c24a0, 0xc0000a7680, 0xc000909b98, 0x5426f20, 0xc000625ba0)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/runner.go:48 +0xa7
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doDev(0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0x0, 0x0)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/dev.go:71 +0x15f
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.doDebug(0x58c24a0, 0xc0000a7680, 0x5880bc0, 0xc000010018, 0xc000463d40, 0x1d)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/debug.go:47 +0xa4
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.cancelWithCtrlC.func1(0x5880bc0, 0xc000010018, 0x0, 0x0)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/signals.go:33 +0xe1
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app/cmd.(*builder).NoArgs.func1(0xc000535180, 0xc00030c000, 0x0, 0xd, 0x0, 0x0)
        /Users/balintp/skaffold/cmd/skaffold/app/cmd/commands.go:96 +0x6a
github.com/spf13/cobra.(*Command).execute(0xc000535180, 0xc000259e10, 0xd, 0xd, 0xc000535180, 0xc000259e10)
        /Users/balintp/skaffold/vendor/github.com/spf13/cobra/command.go:826 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc000534000, 0xc000010018, 0x5880bc0, 0xc000010020)
        /Users/balintp/skaffold/vendor/github.com/spf13/cobra/command.go:914 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
        /Users/balintp/skaffold/vendor/github.com/spf13/cobra/command.go:864
github.com/GoogleContainerTools/skaffold/cmd/skaffold/app.Run(0x5880bc0, 0xc000010018, 0x5880bc0, 0xc000010020, 0xc000625f50, 0x4007f5f)
        /Users/balintp/skaffold/cmd/skaffold/app/skaffold.go:27 +0x57
main.main()
        /Users/balintp/skaffold/cmd/skaffold/skaffold.go:30 +0x4e
 !  minikube  ~/skaf
```

</details>

### PR DESCRIPTION
**Pull request by briandealwis**
Fixes #3821 

**Description**
The `nodejs` debug transformer's [`IsApplicable()` used the presence of `NODE_VERSION`](https://github.com/GoogleContainerTools/skaffold/blob/201c6154105e02b7720099143a9e3cf849acee43/pkg/skaffold/debug/transform_nodejs.go#L59-L61) to indicate that the container image is NodeJS-based.  But the `Apply()` only handles situations where there is a `npm` or `node` invocation.

**User facing changes (remove if N/A)**

_Look ma, no more panics!_

**Follow-up Work (remove if N/A)**

#2141 must be solved to allow running `examples/nodejs` for example.

### CODE REVIEW

File: pkg/skaffold/debug/transform_nodejs.go [hunk 1]
```diff
@@ -206,3 +205,19 @@ func rewriteNpmCommandLine(commandLine []string, spec inspectSpec) []string {
 	}
 	return commandLine
 }
+
+func rewriteNodeOptions(env map[string]string, spec inspectSpec) []v1.EnvVar {
+	found := false
+	var vars []v1.EnvVar
+	for k, v := range env {
```
Comment (loosebazooka) [inline pkg/skaffold/debug/transform_nodejs.go]: can we do something like
```
if val, ok := env["NODE_OPTIONS"]; ok {
    env["NODE_OPTIONS"] = val + " "  + spec.String()
}
else {
    env["NODE_OPTIONS"] = spec.String()
}
```

Comment (loosebazooka) [reply pkg/skaffold/debug/transform_nodejs.go]: oh you have to transform the map...

File: pkg/skaffold/debug/transform_nodejs.go [hunk 2]
```diff
@@ -96,8 +96,7 @@ func (t nodeTransformer) Apply(container *v1.Container, config imageConfiguratio
 			container.Args = rewriteNpmCommandLine(config.arguments, *spec)
 
 		default:
-			logrus.Warnf("Skipping %q as does not appear to invoke node", container.Name)
-			return nil
+			container.Env = rewriteNodeOptions(config.env, *spec)
```
Comment (loosebazooka) [inline pkg/skaffold/debug/transform_nodejs.go]: Just curious, at this point, was there already some hint that this container was node? Could I be a clown and have a java application with a `NODE_OPTIONS` ?

Comment (loosebazooka) [reply pkg/skaffold/debug/transform_nodejs.go]: you know what never mind, I read the description of the PR... :disappointed:

Comment (briandealwis) [reply pkg/skaffold/debug/transform_nodejs.go]: It should be unusual (and discouraged) to have a mega-container with both a JVM and NodeJS application.  In the k8s sphere, they should be separate containers within the same pod.

**Pull request merged**